src/common/transactions/transaction-utils.ts

```diff
@@ -119,3 +128,70 @@ export const generateSTXTransferTx = ({
 
 export const stacksTransactionToHex = (transaction: StacksTransaction) =>
   `0x${transaction.serialize().toString('hex')}`;
+
+export const isContractCallWithTokenTransfers = (tx: Tx) => {
+  return (
+    tx.tx_type === 'contract_call' &&
+    tx.contract_call.function_name === 'transfer' &&
+    tx.contract_call?.function_args?.[0]?.name === 'amount'
+  );
+};
+
+// calculate the real amount of the token based on the decimal number
+// specified in the corresponding token smart contract
+export const calculateTokenTransferAmount = (
+  asset: AssetWithMeta | undefined,
+  tx: ContractCallTransaction
+) => {
+  if (!asset || !asset.meta) return;
+  if (!isContractCallWithTokenTransfers(tx) || !tx.contract_call?.function_args) return;
+  const val = tx.contract_call.function_args[0].repr.replace('u', '');
+  const x = new BigNumber(val);
+  if (!x.isFinite()) return;
+  // Get the decimal number for this token
+  const y = new BigNumber(asset.meta.decimals);
+  // Exponentiation is unsafe here based on SIP-10 and clarity uint 128bits
+  // https://github.com/stacksgov/sips/blob/main/sips/sip-010/sip-010-fungible-token-standard.md
+  const decimalDivisor = new BigNumber(10).pow(y);
+  if (!decimalDivisor.isFinite()) return;
+  return x.dividedBy(decimalDivisor);
```

Comment:
why not use the existing util? https://github.com/blockstack/stacks-wallet-web/blob/main/src/common/stacks-utils.ts#L35

Reply:
I want to still have a BigNumber as return value, but I added the formatting [here](https://github.com/blockstack/stacks-wallet-web/blob/feat/show-contract-activity/I1460/src/store/transactions/transaction.hooks.ts#L279) now.

Reply:
What happens if I want to consume this value without the formatting?